src/main/java/com/autotune/analyzer/recommendations/engine/RecommendationEngine.java

```diff
@@ -200,6 +200,27 @@ public String validate() {
         return validationFailureMsg;
     }
 
+    public String validate_local() {            //TODO Instead of relying on the 'local=true' check everywhere, aim to avoid this complexity by introducing a higher-level abstraction in the code.
+
+        String validationFailureMsg = "";
+        // Check if experiment_name is provided
+        if (experimentName == null || experimentName.isEmpty()) {
+            validationFailureMsg += AnalyzerErrorConstants.APIErrors.UpdateRecommendationsAPI.EXPERIMENT_NAME_MANDATORY + ", ";
+        }
+
+        // Check if interval_end_time is provided
+        if (intervalEndTimeStr != null) {
```

Comment:
This is a redundant check, `isValidDate()` is already checking the `null` case.

Reply:
I removed and tested without passing interval_end_time and it gives me this error 

"message": "Given timestamp - \" null \" is not a valid timestamp format",